### PR TITLE
Add curriculum to course instance pages

### PR DESCRIPTION
**Pull request by ric9176**
After a design/dev session looking at the analytics that students want to see the curriculum on the instance pages, this PR adds them. 

This implied 3 separate layouts for the the curriculum so I abstracted the logic into a function to make the refactor faster and avoid logic repetition in every curriculum component. 

A more extensive refactor here would be to totally separate the data from the UI but there was a time constraint so this could be considered a "half-way" refactor that solves the issue but it should be looked at further at a later time.

### CODE REVIEW

File: src/components/curriculum/CurriculumAdvancedReact.js [hunk 1]
```diff
@@ -96,22 +96,7 @@ const CurriculumAdvancedReact = ({
...
-          <Col md={6} lg={5} lgOffset={1}>
-            {firstHalf}
-            {secondHalf}
-          </Col>
-          <Col md={6} lg={5} lgOffset={1}>
-            <UpcomingTrainingSection curriculum />
-          </Col>
-        </Row>
-      )}
+      {selectCurriculumLayout(firstHalf, secondHalf, layout)}
```
Comment (alexlbr) [inline src/components/curriculum/CurriculumAdvancedReact.js]: @ric9176 I normally go for an object as argument and deconstruct them so I get named arguments which make function's arguments more flexible. But it's fine with me if you prefer "classic" arguments.

File: src/components/curriculum/CurriculumReactNative.js [hunk 2]
```diff
@@ -3,18 +3,19 @@ import { H1Ref } from '../text'
...
 import { REACT_NATIVE } from '../../config/data'
+import selectCurriculumLayout from './selectCurriculumLayout'
 
 const CurriculumReactNative = ({
   showTitle = true,
-  list,
   enableToggle,
   isOpen,
   toggleNavigateTo = `/curriculum?tab=${REACT_NATIVE}`,
+  selectLayout,
```
Comment (alexlbr) [inline src/components/curriculum/CurriculumReactNative.js]: @ric9176 what is this prop for?

Comment (ric9176) [reply src/components/curriculum/CurriculumReactNative.js]: :+1: I will remove that in tidy up!

File: src/components/curriculum/CurriculumAdvancedReact.js (hunk 1 above)
Comment (ric9176) [reply src/components/curriculum/CurriculumAdvancedReact.js]: :+1: I'll refactor, this is a better approach as in the reactNative there is no secondHalf so I can avoid passing a null value :smile:

**Pull request merged**